openstack/swift/bin/haproxy-start

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+CFG_DIR=/usr/local/etc/haproxy
+cp -f /haproxy-etc-cluster/haproxy.cfg ${CFG_DIR}/haproxy.cfg
+cp -f /haproxy-etc-cluster/dhparam.pem ${CFG_DIR}/dhparam.pem
+
+SSL_DIR=${CFG_DIR}/ssl
+mkdir -p ${SSL_DIR}
+cp -f /tls-secret/tls.crt ${SSL_DIR}
+cp -f /tls-secret/tls.key ${SSL_DIR}
+cat /tls-secret/tls.crt /tls-secret/tls.key > ${SSL_DIR}/tls.pem
+
+# in a subshell, run a recurring check for new TLS secrets
+( while true; do
+
+  sleep 30
+
+  # do not attempt to install new cert if the certs are broken (e.g. because we catch
+  # k8s in the middle of mounting a new secret)
+  if ! grep -q 'END CERTIFICATE' /tls-secret/tls.crt; then
+    echo "$(date): cert watch: skipping reload because /tls-secret/tls.crt is broken"
+    continue
+  fi
+  if ! grep -q 'END RSA PRIVATE KEY' /tls-secret/tls.key; then
+    echo "$(date): cert watch: skipping reload because /tls-secret/tls.key is broken"
+    continue
+  fi
+
+  # only reload if secrets have changed
+  if diff -q /tls-secret/tls.crt ${SSL_DIR}/tls.crt && diff -q /tls-secret/tls.key ${SSL_DIR}/tls.key; then
+    : # nothing to do
+  else
+    cp -f /tls-secret/tls.crt ${SSL_DIR}
+    cp -f /tls-secret/tls.key ${SSL_DIR}
+    cat /tls-secret/tls.crt /tls-secret/tls.key > ${SSL_DIR}/tls.pem
+    echo "$(date): cert watch: reloading haproxy"
+    # TODO: This needs revisting. Don't kill all haproxy at once. This is not a reload
+    kill -USR1 $(pidof haproxy)
```

Comment:
I would like to have a random sleep here to ensure that, when a cert change happens, the haproxy instances don't all restart at the same time:

```suggestion
    sleep $(( RANDOM % 300 ))
    kill -USR1 $(pidof haproxy)
```